src/syntax/read.rkt

```diff
@@ -147,7 +147,22 @@
   ;; inline and desugar
   (define body* (fpcore->prog body ctx))
   (define pre* (fpcore->prog (dict-ref prop-dict ':pre 'TRUE) ctx))
-  (define target (fpcore->prog (dict-ref prop-dict ':alt #f) ctx))
+
+  (define targets
+    (for/list ([(key val) (in-dict prop-dict)] #:when (eq? key ':alt))
+      (define plat-name (extract-platform-name val))  ; plat-name is symbol or #f
+
+      (cond
+        ; If plat-name extracted, check if name matches
+        [plat-name (cons val (equal? plat-name (*platform-name*)))]
+
+        ; try to lower
+        [else
+          (with-handlers ([exn:fail:user:herbie:missing? (lambda (e) (cons val #f))])
+            ; Testing if error thrown
+            (spec->prog val ctx)
+            (cons val #t))])))
```

Comment:
```suggestion
      (match (extract-platform-name val)  ; plat-name is symbol or #f
        ; If plat-name extracted, check if name matches
        [(? symbol? plat-name) (cons val (equal? plat-name (*platform-name*)))]

        ; try to lower
        [#f
          (with-handlers ([exn:fail:user:herbie:missing? (lambda (e) (cons val #f))])
            ; Testing if error thrown
            (spec->prog val ctx)
            (cons val #t))])))
```

Reply:
With the new change, the `(? symbol? ...)` is way cleaner